File: tests/go/cases/case_278_thread_pool_capture_07.go
package escape_tests

var retainedCase278 = []map[string]string{}

func Case278ThreadPoolCapture07(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_pool_capture_07",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "thread_pool_capture_07:" + raw,
	}
	// ESCAPE: goroutine captures payload and retains it.
	go func(v map[string]string) { retainedCase278 = append(retainedCase278, v) }(payload)
	return "ok"
}

File: tests/nodejs/cases/case_278_thread_pool_capture_07.js
/**
 * Task case 278: thread_pool_capture_07 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case278ThreadPoolCapture07(input) {
  const taskName = 'thread_pool_capture_07';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async task captures payload.
  setTimeout(() => retainedAudit.push(payload), 0);
  return 'ok';
}

module.exports = { case278ThreadPoolCapture07 };
